### DISCUSSION TITLE
Difficult to use~

### DISCUSSION THREAD

**Discussion by vansin** [open-mmlab/mim#165] [Announcements]
![image](https://user-images.githubusercontent.com/25839884/192153705-802f40c9-00b9-4b1d-a0e6-982140454f94.png)

![image](https://user-images.githubusercontent.com/25839884/192153843-dd7a782b-966a-467c-b4ab-6f1122918fcf.png)

**Comment by mm-assistant** [comment]
We recommend using English or English & Chinese for issues so that we could have broader discussion.

**Comment by ice-tong** [comment]
感谢反馈！

mim install 相对于 pip install 的变化有两点：
- 自动构造出 mmcv 的预编译包下载链接
- 自动处理和安装下游算法库对上游算法库以及 mmcv 的依赖

“自定安装依赖”指的是 mim install 会“自动处理安装对上游算法库和 mmcv 的依赖”吗，如果担忧的是这个问题的话，推荐只使用 mim install mmcv 即可。

在一个环境中，如果需要同时安装多个算法库，这些算法库对 mmcv 的版本依赖不同，这个时候使用 mim 来安装这些算法库，则可以自动的找到一个最合适的 mmcv 版本进行安装。

**Comment by Pillar1989** [comment]
直接用 pip install 不好么 ？ 造这个轮子的意义是啥  ？  @ice-tong

**Comment by ice-tong** [comment]
> 直接用 pip install 不好么 ？ 造这个轮子的意义是啥 ？ @ice-tong

哈喽 @Pillar1989 ，感谢对 mim 的关注～

mim install 只是 mim 其中一个功能，mim 除了提供包安装功能之外，还提供预训练模型查询下载功能，和各算法库脚本统一调用功能。

至于 mim install，它的存在主要解决了两个问题：
- 早期 OpenMMLab 算法库打包发布机制尚未健全，通过 pip install 无法安装无法获取模型管理和脚本调用所需的资源文件，因此通过 mim install 来应对此问题。
- pip install 暂时无法从 pypi 上下载对应环境所需要的预编译包，需要下载源码编译或者手动构造预编译包链接。mim install 则会自动的根据当前环境构造预编译包链接，并下载安装。

mim 不止于 mim install 一个功能，mim install 从现阶段来看也有其存在的意义，所以 mim 并不是重复造轮子的工作。 ^_^
